apps/rush-lib/src/cli/actions/PublishAction.ts

```diff
@@ -453,4 +472,28 @@ export class PublishAction extends BaseRushAction {
 
     return `${name}-${project.packageJson.version}.tgz`;
   }
-}
\ No newline at end of file
+
+  private _manageLooseVersions(action: 'lock' | 'unlock'): void {
```

Comment:
Can you refactor this into two functions? Maybe each can call a helper `processAllProjectDependencies(() => { ... })` or something?

The `action: 'lock' | 'unlock'` parameter feels awkward.